### ISSUE TITLE
resource/aws_elasticache_replication_group, data_source/aws_elasticache_replication_group: Remove deprecated parameters

### ISSUE THREAD

**Issue by anGie44** [hashicorp/terraform-provider-aws#22903]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The resource and data source `aws_elasticache_replication_group` have a number of deprecated parameters introduced as part of https://github.com/hashicorp/terraform-provider-aws/issues/18440. They should be removed in the next major release.

Resource:
* `number_cache_clusters`
* `cluster_mode`
* `replication_group_description`

Data Source:
* `number_cache_clusters`
* `replication_group_description`

**Comment by rquadling** [comment]
Having an issue with these deprecated elements in that whilst the `terraform plan` output shows a warning
```
╷
│ Warning: Argument is deprecated
│ 
│   with aws_elasticache_replication_group.redis_sharded,
│   on [elasticache.tf](https://bitbucket.org/digitickets/inf-store/src/38a5d22d4f3bcfe70349478c299a38b9a8ed8174/elasticache.tf) line 42, in resource "aws_elasticache_replication_group" "redis_sharded":
│   42: resource "aws_elasticache_replication_group" "redis_sharded" {
│ 
│ Use num_node_groups and replicas_per_node_group instead
│ 
│ (and 4 more similar warnings elsewhere)
╵
```

Which isn't that useful as what are the other 4.

In addition, `terraform validate` isn't outputting these. Instead, I'm getting
```json
{
  "format_version": "1.0",
  "valid": true,
  "error_count": 0,
  "warning_count": 0,
  "diagnostics": []
}
```

I've raised a question at https://discuss.hashicorp.com/t/where-are-warning-argument-is-deprecated-present-in-the-plan/47581 also to see if there's something I'm missing.

As an aside, the additional reporting we have in our pipeline has correctly shown up all the S3 related deprecations correctly and allowed us to successfully migrate all the deprecated `aws_s3_buckets` elements to their new resource counterparts.

So we assumed that the warnings that `terraform plan` is outputting would be present in the `terraform validate -json` output (just like the S3 ones where).

**Comment by jar-b** [comment]
Closed by #31008, merged to main via #31392
